Feature/pfg5 50 - Validation for Category

RominaMusacchio opened the pull request:
Adding DTO and validations for Category following previous GlobalHandlerException.

Name: 3<x<32
Description: 15<x<256

https://github.com/user-attachments/assets/dda62def-ae8c-4100-816d-a0645430b54f

RominaMusacchio committed:
Validations for Category added to service level.

Name: 3<x<32
Description: 5<x<256

RominaMusacchio committed:
Merge branch 'develop' of https://github.com/strikerout/CAPITRAVEL_BE into feature/PFG5-50

RominaMusacchio committed:
Validations for Category with DTO.
Name: 3<x<32
Description: 5<x<256

strikerout commented on src/main/java/com/capitravel/Capitravel/service/impl/CategoryServiceImpl.java:
```diff
@@ -21,7 +23,13 @@ public CategoryServiceImpl(CategoryRepository categoryRepository) {
     @Override
     public Category getCategoryById(Long id) {
-        return categoryRepository.findById(id).orElse(null);
+        Category idCategory = categoryRepository.findById(id).orElse(null);
```
no vas a retornar un Id de categoría, sino una categoría
llamaría a esta variable simplemente category

RominaMusacchio replied:
Done

strikerout commented on src/main/java/com/capitravel/Capitravel/model/Category.java:
```diff
@@ -16,6 +15,7 @@ public class Category {
     @GeneratedValue(strategy = GenerationType.IDENTITY)
     private Long id;
 
+
```
Podemos quitar este espacio agregado? no es necesario

RominaMusacchio replied:
Done

strikerout commented on src/main/java/com/capitravel/Capitravel/service/impl/CategoryServiceImpl.java:
```diff
@@ -21,7 +23,13 @@ public CategoryServiceImpl(CategoryRepository categoryRepository) {
-        return categoryRepository.findById(id).orElse(null);
+        Category idCategory = categoryRepository.findById(id).orElse(null);
+
+        if(idCategory!=null){
```
Intentemos respetar el formateo, un espacio entre variable condición y condicional (category != null)

RominaMusacchio replied:
Done

strikerout commented on src/main/java/com/capitravel/Capitravel/service/impl/CategoryServiceImpl.java:
```diff
@@ -30,25 +38,32 @@ public List<Category> getAllCategories() {
+        category.setName(categoryDTO.getName());
+        category.setDescription(categoryDTO.getDescription());
         return categoryRepository.save(category);
+
```
podemos quitar este espacio?

RominaMusacchio replied:
Done

strikerout commented on src/main/java/com/capitravel/Capitravel/service/impl/CategoryServiceImpl.java:
```diff
@@ -30,25 +38,32 @@ public List<Category> getAllCategories() {
     public void deleteCategory(Long id) {
         categoryRepository.deleteById(id);
     }
+
```
podemos quitar este espacio?

RominaMusacchio replied:
Done

strikerout commented on src/main/java/com/capitravel/Capitravel/service/impl/CategoryServiceImpl.java:
```diff
@@ -30,25 +38,32 @@ public List<Category> getAllCategories() {
-            updatedCategory.setName(category.getName());
-            updatedCategory.setDescription(category.getDescription());
-            return categoryRepository.save(updatedCategory);
+
```
quitamos este espacio?

RominaMusacchio replied:
Done

strikerout commented on src/main/java/com/capitravel/Capitravel/service/impl/CategoryServiceImpl.java:
```diff
@@ -30,25 +38,32 @@ public List<Category> getAllCategories() {
+                Category updatedCategory = existingCategory.get();
+                updatedCategory.setName(categoryDTO.getName());
+                updatedCategory.setDescription(categoryDTO.getDescription());
+                return categoryRepository.save(updatedCategory);
```
todo este bloque está mal identado, debe correrse un tab más a la izquierda

RominaMusacchio replied:
Done

strikerout commented on src/main/java/com/capitravel/Capitravel/service/impl/CategoryServiceImpl.java:
```diff
@@ -30,25 +38,32 @@ public List<Category> getAllCategories() {
+                updatedCategory.setName(categoryDTO.getName());
+                updatedCategory.setDescription(categoryDTO.getDescription());
+                return categoryRepository.save(updatedCategory);
+
```
quitamos este espacio?

RominaMusacchio replied:
Done

strikerout commented on src/main/java/com/capitravel/Capitravel/dto/CategoryDTO.java:
```diff
@@ -0,0 +1,17 @@
+    @NotBlank(message = "Description is required")
+    @Size(min= 15,max = 256, message = "Description must be between 15 and 256 characters")
+    private String description;
+
```
No dejemos espacios antes del final de la clase, podemos quitarlo?

RominaMusacchio replied:
Done

RominaMusacchio committed:
Made changes for PR.
Deleted spaces, fixed indentation where required, changed idCategory for category on CategoryServiceImpl > "getCategoryById"
